src/main/java/com/hubspot/jinjava/interpret/NullValue.java

```diff
@@ -0,0 +1,21 @@
+package com.hubspot.jinjava.interpret;
+
+/**
+ * Marker object of a `null` value. A null value in the map is usually considered
+ * the key does not exist. For example map = {"a": null}, if map.get("a") == null,
+ * we treat it as the there is not key "a" in the map.
+ */
+public final class NullValue {
+  public static final NullValue INSTANCE = new NullValue();
```

Comment:
Just having this have the String value of `null` wouldn't be sufficient.
Likely you'd want to unwrap the `NullValue -> null` inside of `com.hubspot.jinjava.util.ScopeMap#get(Object)`

Creating some construct such as:
```java
record ScopeMap.ScopeValue(Object value) {}
```
And always unwrapping those values when accessing them inside of a ScopeMap seems like it would do the trick

Reply:
Can you elaborate why we need to unwrap the NullValue -> null? I don't think we need anywhere to use the `null` value.

Reply:
`null != "null"`
Here's a quick example:
```
{% set foo = [0, 1, None, 2] %}
{% for i in foo %}
{% if i %}
 i is {{ i }}
{% endif %}
{% endfor %}
```
The expected jinja output is:
```
 i is 1
 i is 2
```
If treating `null` as `"null"`, the output would be:
```
i is 1
i is null
i is 2
```